src/com/google/javascript/jscomp/parsing/parser/Parser.java

```diff
@@ -2824,7 +2824,7 @@ private ParseTree parseAssignment(Expression expressionIn) {
 
     if (isStartOfAsyncArrowFunction(left)) {
       // re-evaluate as an async arrow function.
-      resetScanner(left);
+      resetScanner(start);
```

Comment:
@brad4d Your mental model was completely correct - but this PR changes what `start` refers to.